pkg/heartbeat/heartbeat.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package heartbeat
```

Comment:
You could use a filtered `/components` list to retrieve known apiservers, but you'd still need a separate etcd key/value for synchronizing leader election. You wouldn't be able to use a field on the Components api. That said, we could add a field on the Components api that is auto-populated on read from some other key/value. You just wouldn't be able to write to it. I'm not sure if that's useful tho. It's probably better to read from the source of truth and piece them together on the client side.